src/http/api/account.ts

```diff
@@ -39,75 +43,37 @@ type FollowAccount = Pick<
  *
- * @param accountService Account service instance
+ * @param db Database client
+ * @param fedifyContextFactory Fedify context factory instance
  */
 export function createGetAccountHandler(
-    accountService: AccountService,
-    accountRepository: KnexAccountRepository,
-    fedify: Federation<ContextData>,
+    db: Knex,
+    fedifyContextFactory: FedifyContextFactory,
 ) {
     /**
      * Handle a request for an account
      *
      * @param ctx App context
      */
     return async function handleGetAccount(ctx: AppContext) {
-        const logger = ctx.get('logger');
-        const site = ctx.get('site');
-        let account: Account | null = null;
-        const db = ctx.get('db');
-
-        const apCtx = fedify.createContext(ctx.req.raw as Request, {
-            db,
-            globaldb: ctx.get('globaldb'),
-            logger,
-        });
-
-        const defaultAccount = await accountRepository.getBySite(
-            ctx.get('site'),
-        );
-
         const handle = ctx.req.param('handle');
-        // We are using the keyword 'me', if we want to get the account of teh current user
-        if (handle === 'me') {
-            account = defaultAccount;
-        } else {
-            if (!isHandle(handle)) {
-                return new Response(null, { status: 404 });
-            }
 
-            const apId = await lookupAPIdByHandle(apCtx, handle);
-            if (apId) {
-                account = await accountRepository.getByApId(new URL(apId));
-            }
+        if (handle !== CURRENT_USER_KEYWORD && !isHandle(handle)) {
+            return new Response(null, { status: 404 });
         }
 
-        let accountDto: AccountDTO;
+        let accountDto: AccountDTO | null = null;
 
-        try {
-            //If we found the account in our db and it's an internal account, do an internal lookup
-            if (account?.isInternal) {
-                accountDto = await getAccountDTOFromAccount(
-                    account,
-                    defaultAccount,
-                    accountService,
-                );
-            } else {
-                //Otherwise, do a remote lookup to fetch the updated data
-                accountDto = await getAccountDTOByHandle(
-                    handle,
-                    apCtx,
-                    site,
-                    accountService,
-                );
-            }
-        } catch (error) {
-            logger.error('Error getting account: {error}', { error });
+        const accountView = new AccountView(db, fedifyContextFactory);
 
-            return new Response(null, { status: 500 });
+        if (handle === CURRENT_USER_KEYWORD) {
+            accountDto = await accountView.viewBySite(ctx.get('site'));
+        } else {
+            accountDto = await accountView.viewByHandle(handle, {
+                site: ctx.get('site'),
+            });
         }
 
-        // Return response
         return new Response(JSON.stringify(accountDto), {
```

Comment:
We don't check for the `accountDto` being `null` here - I think we wanna 404 in that case probs